press/views/legacy/redirects.py

```diff
@@ -0,0 +1,39 @@
+from pyramid import httpexceptions
+from pyramid.view import view_config
+
+from press.utils import convert_to_legacy_domain
+
+
+def _lookup_latest_legacy_version(request, legacy_id):
+    """Given the legacy id for content, lookup the latest version."""
+    t = request.db_tables
+    with request.get_db_engine('common').begin() as db_conn:
+        results = db_conn.execute(
+            t.latest_modules.select()
```

Comment:
This does not take into account the problem I mentioned in scrum yesterday: what if a book is published that breaks baking of that content? That version will exist, but _not_ be considered the latest.
So, the query to get the version with the largest published version has to do the work of extracting it from modules, considering the major and minor version numbers. Cribbed from the `current_modules` view, but removing the state parts:
```
SELECT moduleid, version FROM modules m2 
                WHERE m2.moduleid = %s 
                    AND m2.major_version = (
                    SELECT max(major_version) FROM modules m3
                        WHERE m2.uuid = m3.uuid
                )
                AND
                (m2.minor_version IS NULL OR
                 m2.minor_version = (
                    SELECT max(minor_version) FROM modules m4
                        WHERE m2.uuid = m4.uuid AND
                        m2.major_version = m4.major_version
                    )
            ) order by moduleid;
```

Reply:
Looks a bit scary, but executes in < 5ms (because of indexes)